README.md

```diff
@@ -74,39 +44,176 @@ This program was created with the following tech stack and packages:
-   operating system. We recommend the Powershell approach for Windows users.
-7. Run `npm install`
-8. Run `npm start`
+10. Use Postman or a similar platform to send a POST request (see [POST Request Properties](#post-request-properties))
+to the local API that is currently listening for a request from step 7.
+11. Wait for the API to process the request and return the video
+12. Enjoy the animation!
 
 ## How it Works
 
-The Kurkarella TTS app creates the synthesized speech file from the user’s voice and text input. The Animation Add-On
-is broken down into two main components: the phoneme factory and the animation processor. The phoneme factory is in
-charge of mapping spoken language from an audio file into a series of phonemes, which are units of sound that
-distinguish one word from another. In order to make this process as accurate as possible, the phoneme factory also
-receives the user text file and language selection.
+### Overview
+
+The Speaking Portal Project (SPP) is built to connect with Kukurella's Text-to-Speech platform as an API. The SPP is
+broken down into three main components: **The API**, **The Phoneme Factory**, and **The Animation Factory**.
+
+The Speaking Portal API generates an MP4 animation from the speech file and text file received from Kukarella's
+TTS process. Text and audio inputs are first processed and converted into a json file containing phonemes and their
+timings. This processing is done via the [Rhubarb library](https://github.com/DanielSWolf/rhubarb-lip-sync). (Phonemes
+are the smallest unit of sounds that uniquely identify distinct sounds within words.) Once a phoneme file is created,
+the animation process then identifies which avatar assets to use based on the phoneme. Other features, such as blinks
+and pose changes are added, and then all image assets are rendered in a video file through ffmepg.
+
+### API
+
+When the API receives a request, the uploaded files are stored in the `/tmp` directory. The API code then reads these inputs and converts them to the appropriate format for Rhubarb and ffmpeg. Upon conversion, the animation process is started by sending all user inputs to the `main.ts` file. 
+
+Here's a typical example of the JSON sent in a POST request.
+
+```markdown
+{
+  fieldname: 'audio',
+  originalname: 'en-Amber.wav',
+  encoding: '7bit',
+  mimetype: 'audio/wave',
+  destination: './tmp',
+  filename: 'f8fc3bc5c3f0fbc70ab0899d0dc0c860',
+  path: 'tmp\\f8fc3bc5c3f0fbc70ab0899d0dc0c860',
+  size: 9049278
+} {
+  fieldname: 'text',
+  originalname: 'en-text.txt',
+  encoding: '7bit',
+  mimetype: 'text/plain',
+  destination: './tmp',
+  filename: 'e05b5794e54ca3810b071b87bc1449e1',
+  path: 'tmp\\e05b5794e54ca3810b071b87bc1449e1',
+  size: 1363
+} phonetic
+```
+
+#### POST Request Properties
+
+The SPP API is the first step in the animation process. Kukurella requests are sent to the SPP API as a POST with the
```

Comment:
```suggestion
Kukurella requests are sent to the SPP API as a POST with the
```
Technically not part of the animation process and we say below that the animation factory is the first step